kafka_consumer/tests/python_client/test_unit.py

```diff
@@ -134,79 +135,99 @@ def test_tls_config_legacy(extra_config, expected_http_kwargs, check, kafka_inst
         ),
         pytest.param(
             {'kafka_connect_str': None},
-            pytest.raises(ConfigurationError, match='kafka_connect_str should be string or list of strings'),
+            pytest.raises(Exception, match='kafka_connect_str\n  none is not an allowed value'),
+            '',
             0,
             id="Invalid Nonetype kafka_connect_str",
         ),
         pytest.param(
             {'kafka_connect_str': [KAFKA_CONNECT_STR, '127.0.0.1:9093'], 'monitor_unlisted_consumer_groups': True},
             does_not_raise(),
+            '',
             4,
             id="Valid list kafka_connect_str",
         ),
         pytest.param(
             {'kafka_connect_str': KAFKA_CONNECT_STR, 'monitor_unlisted_consumer_groups': True},
             does_not_raise(),
+            '',
             4,
             id="Valid str kafka_connect_str",
         ),
-        pytest.param({'kafka_connect_str': 'invalid'}, pytest.raises(Exception), 0, id="Invalid str kafka_connect_str"),
+        pytest.param(
+            {'kafka_connect_str': 'invalid'},
+            pytest.raises(Exception),
+            'ConfigurationError: Cannot fetch consumer offsets because no consumer_groups are specified and '
+            'monitor_unlisted_consumer_groups is False',
+            0,
+            id="Invalid str kafka_connect_str",
+        ),
         pytest.param(
             {'kafka_connect_str': KAFKA_CONNECT_STR, 'consumer_groups': {}, 'monitor_unlisted_consumer_groups': True},
             does_not_raise(),
+            '',
             4,
             id="Empty consumer_groups and monitor_unlisted_consumer_groups true",
         ),
         pytest.param(
             {'kafka_connect_str': KAFKA_CONNECT_STR, 'consumer_groups': {'my_consumer': None}},
             does_not_raise(),
+            '',
             4,
             id="One consumer group, all topics and partitions",
         ),
         pytest.param(
             {'kafka_connect_str': KAFKA_CONNECT_STR, 'consumer_groups': {'my_consumer': {'marvel': None}}},
             does_not_raise(),
+            '',
             2,
             id="One consumer group, one topic, all partitions",
         ),
         pytest.param(
             {'kafka_connect_str': KAFKA_CONNECT_STR, 'consumer_groups': {'my_consumer': {'marvel': [1]}}},
             does_not_raise(),
+            '',
             1,
             id="One consumer group, one topic, one partition",
         ),
     ],
 )
-def test_config(check, instance, aggregator, expected_exception, metric_count):
+def test_config(dd_run_check, check, instance, aggregator, expected_exception, exception_msg, metric_count, caplog):
+    caplog.set_level(logging.DEBUG)
     with expected_exception:
-        check(instance).check(instance)
+        dd_run_check(check(instance, client=KafkaPythonClient))
 
     for m in metrics:
         aggregator.assert_metric(m, count=metric_count)
 
+    assert exception_msg in caplog.text
```

Comment:
This looks kind of weird, but some exception messages [don't get raised],(https://github.com/DataDog/integrations-core/blob/9f245a4b497d166815fced239f0eb34e5a6c3890/kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py#L42) so `pytest.raises()` doesn't catch them. caplog is being used here to catch those messages in the logs.